### PR TITLE
Add Docker image

### PR DESCRIPTION
**Pull request by mgcrea**
Adds a working `Dockerfile` to generate a ready to use Docker image.

Ideally you would have a CI job that does build the image and push automatically on success. If interested I could send another PR doing just that using GitHub actions / CircleCI.

### CODE REVIEW

**Comment by hwwhww** [comment]
Hi @mgcrea! It looks good and addresses #22. 👍 
Do you mind changing the target branch to `dev`?

**Comment by mgcrea** [comment]
@hwwhww thanks! Just changed the target branch, looks like there is no conflicts.

**Comment by ltfschoen** [comment]
How about a .dockerignore file
```
# Prevent your local modules and debug logs from being
# copied onto your Docker image and possibly overwriting
# modules installed within your image
.git
__pycache__
```

File: README.md [hunk 1]
```diff
@@ -201,6 +201,28 @@ See [here](#arguments)
...
+```sh
+docker run -it --rm -v $(pwd)/validator_keys:/app/validator_keys ethereum/eth2.0-deposit-cli
+```
+
+You can also run the tool with optional arguments:
+
+```sh
+docker run -it --rm -v $(pwd)/validator_keys:/app/validator_keys ethereum/eth2.0-deposit-cli --num_validators=<NUM_VALIDATORS> --mnemonic_language=english --folder=<YOUR_FOLDER_PATH>
+```
+
```
Comment (ltfschoen) [inline README.md:219]: Suggest adding an example of what to run for 1 validator on the Medalla testnet using English

e.g.
```
docker run -it --rm -v $(pwd)/validator_keys:/app/validator_keys ethereum/eth2.0-deposit-cli --num_validators=1 --mnemonic_language=english --chain=medalla
```

**Comment by mgcrea** [comment]
@ltfschoen Thanks for the review+suggestions, just pushed a new commit including them.

**Comment by PandaWhisperer** [comment]
Just spent the better part of an hour trying to make a `Dockerfile` myself. Finally got it working, then I come here to see that someone already thought of it... 😅 

Also, this version is probably better because it produces a smaller image (mine was almost 1GB because I didn't know the right packages to install, so I used buster instead of alpine). 

Just tested this PR and seems to work fine for me. 

One suggestion I have is perhaps using a [multi-stage build](https://docs.docker.com/develop/develop-images/multistage-build/) to make the final image size even smaller, so it can be more easily distributed (say, via Docker Hub). 

I was able to reduce the final image size by about 220 MB by adding the following lines to the `Dockerfile`:

```
FROM python:3.7-alpine

WORKDIR /app

COPY --from=0 /app /app

ENTRYPOINT [ "python3", "./eth2deposit/deposit.py" ]
```

All this does is basically start over with a new image, copy the files from the build, and then package that up, so the final image does not include the compiler and libraries that were necessary to build the app.

**Comment by ltfschoen** [comment]
Another useful resource is this https://github.com/stefa2k/prysm-docker-compose

**Comment by PandaWhisperer** [comment]
@ltfschoen that setup is what I use to run my validators for Medalla and also the reason why I wanted to dockerize the deposit CLI. 

Stefan has done a great job making the key import as simple as possible, but it still took a bit of work to get the deposit CLI working, so I thought if only there was a docker image for that, this could literally become a one stop shop for setting up a validator, no futzing around with different compilers etc. required. 

Unfortunately I did not find your PR there, @ltfschoen?

@mgcrea I think one improvement that would be nice to have would be a smaller image size. Yours was 264 MB, which is already a huge improvement over my 957, but I think it could be trimmed down quite a bit using a [multi-stage build](https://docs.docker.com/develop/develop-images/multistage-build/), which could basically drop the compiler and devtools required for the build process and only contain what's necessary to run the actual CLI.

I tried my hand at that, but unfortunately I failed, because my image did not include the compiled eggs, so it wouldn't run.

**Comment by ltfschoen** [comment]
@PandaWhisperer FYI this is the PR with what I did https://github.com/ethereum/eth2.0-deposit-cli/pull/83

**Comment by PandaWhisperer** [comment]
@ltfschoen oh okay, I misunderstood you then. I thought you made a PR on stefa2k/prysm-docker-compose.

**Comment by mgcrea** [comment]
Is there anything left I can do to help this being merged? Thanks!

File: README.md [hunk 2]
```diff
@@ -201,6 +201,34 @@ See [here](#arguments)
 ###### Successful message
 See [here](#successful-message)
 
+#### Option 4. Use Docker image
+
+##### Step 1. Create keys and `deposit_data-*.json`
```
Comment (hwwhww) [inline README.md:206]: Should the docs tell the user to build the docker image first? (`make build_docker`)

**Review by hwwhww — APPROVED** [review]
Since we (EF) may not upload the docker image somewhere officially, I assume that the users have to build the docker image by themselves. That is, we probably should tell them how to build it in the document.

Otherwise, this PR looks good to me. Thank you! 😊

/cc @CarlBeek

**Comment by hwwhww** [comment]
Hey @mgcrea 👋 are you still working on it?

**Comment by mgcrea** [comment]
Hey @hwwhww, yes will gladly add any changes required for this to land. I've added the `make build_docker` step.

**Review by hwwhww — APPROVED** [review]
LGTM!

/cc @CarlBeek would you like to take a look at it before merge?

**Comment by CarlBeek** [comment]
Looks good to me too. Thanks for your work on this @mgcrea!

**Comment by hwwhww** [comment]
Thank you @mgcrea!

> Ideally you would have a CI job that does build the image and push automatically on success. If interested I could send another PR doing just that using GitHub actions / CircleCI.

Would you like to help with the CI job? #123 will change the command line interface so we should have dockerfile build in CI.

**Pull request merged**